src/api/leaderboards.rs

```diff
@@ -54,111 +65,168 @@ pub async fn create_leaderboard(
     }
 }
 
-#[get("/leaderboards/{name}")]
+#[utoipa::path(
+    get,
+    path = "/leaderboards/{name}",
+    params(
+        ("name", description = "Leaderboard name")
+    ),
+    security(
+        ("api_key" = [])
+    ),
+    responses(
+        (status = OK, body = PrivateLeaderboard)
+    )
+)]
 pub async fn get_leaderboard(
     user: UserId,
-    path: Path<(String,)>,
+    Path(name): Path<String>,
     db: DatabaseWrapper,
-) -> Result<impl Responder, TimeError> {
+) -> Result<Json<PrivateLeaderboard>, TimeError> {
     let lid = db
-        .get_leaderboard_id_by_name(&path.0)
+        .get_leaderboard_id_by_name(&name)
         .await
         .map_err(|_| TimeError::LeaderboardNotFound)?;
 
     if db.is_leaderboard_member(user.id, lid).await? {
-        let board = db.get_leaderboard(&path.0).await?;
-        Ok(web::Json(board))
+        let board = db.get_leaderboard(&name).await?;
+        Ok(Json(board))
     } else {
         Err(TimeError::Unauthorized)
     }
 }
 
-#[delete("/leaderboards/{name}")]
+#[utoipa::path(
+    delete,
+    path = "/leaderboard/{name}",
```

Comment:
Should it be plural instead?
```suggestion
    path = "/leaderboards/{name}",
```